pkg/policy/api/l4.go

```diff
@@ -112,6 +112,39 @@ type TLSContext struct {
 	PrivateKey string `json:"privateKey,omitempty"`
 }
 
+// EnvoyConfig defines a reference to a CiliumEnvoyConfig or CiliumClusterwideEnvoyConfig
+type EnvoyConfig struct {
+	// Kind is the resource type being referred to. Defaults to CiliumEnvoyConfig or
+	// CiliumClusterwideEnvoyConfig for CiliumNetworkPolicy and CiliumClusterwideNetworkPolicy,
+	// respectively. The only case this is currently explicitly needed when referring to a
+	// CiliumClusterwideEnvoyConfig from CiliumNetworkPolicy, as using a namespaced listener
+	// from a cluster scoped policy is not allowed.
+	//
+	// +kubebuilder:validation:Enum=CiliumEnvoyConfig;CiliumClusterwideEnvoyConfig
+	// +kubebuilder:validation:Optional
+	Kind string `json:"kind"`
+
+	// Name is the resource name of the CiliumEnvoyConfig or CiliumClusterwideEnvoyConfig where
+	// the listener is defined in.
+	//
+	// +kubebuilder:validation:Required
+	Name string `json:"name"`
+}
+
+// Listener defines a reference to an Envoy listener specified in a CEC or CCEC resource.
+type Listener struct {
+	// EnvoyConfig is a reference to the CEC or CCNP resource in which
+	// the listener is defined.
+	//
+	// +kubebuilder:validation:Required
+	EnvoyConfig *EnvoyConfig `json:"envoyConfig"`
+
+	// Name is the name of the listener.
+	//
+	// +kubebuilder:validation:Required
+	Name string `json:"name"`
```

Comment:
If you don't expect this to be empty, then we can add validation on length so that it's rejected if empty.

[1]

Reply:
Missed this, thought that would be part of required... ofcause it is not!

Reply:
Added:
```
	// +kubebuilder:validation:MinLength=1
```